### ISSUE TITLE
Add a changelog

### ISSUE THREAD

**Issue by felixfbecker** [Microsoft/language-server-protocol#162]
It is very hard to follow all the recent updates to the spec just through git commit log and inline comments like

> **New:** whatever

those will also clutter the spec more and more over time or have to be removed periodically.
Please add a CHANGELOG.md to document changes and new versions.

**Comment by dbaeumer** [comment]
The way how this works right now is as follows:

- the protocol.md in root is always the latest version. It has only New and Changed for changes from the last major version
- there is a versions folder with the old versions.

So the changelog is basically in lined. I found that more convenient. If you still think that a change log is superior is someone volunteering to maintain one :-)?

**Comment by felixfbecker** [comment]
Like I said in the original post, I think that in the long run that will be hard to maintain. And there are not only major versions, adding a new method for example can just be a minor update to the spec. When as an LS author you come back after a few weeks and want to know what changed, a changelog would be handy. I also think version history would be better managed through git than duplicating files.

I don't think a volunteer could maintain a changelog, the changelog needs to be updated when the change is made by the one making the change. There are some tools though that can generate them from commit messages, but only if you follow certain conventions, I don't think they would be useful here.

It's also not always additions to the protocol. For example, in 3.0 `textEdit` was deprecated and I kept that in mind for my LS, and so did client maintainers. A few commits later it was silently un-deprecated, which caused quite some confusion. It was basically just luck that I noticed this while looking something up in the protocol and first thought it was just a merge that went wrong. A changelog entry would have made this clear.

Thanks for considering!
